dsl/camel-jbang/CamelJbang.java

```diff
@@ -0,0 +1,255 @@
+import org.apache.camel.github.GitHubResourceResolver;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.apache.logging.log4j.core.config.DefaultConfiguration;
+
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Parameters;
+import picocli.CommandLine.Option;
+
+@Command(name = "run", description = "Run a Kamelet")
+class Run implements Callable<Integer> {
+    private CamelContext context;
+
+    @Parameters(description = "The path to the kamelet binding", defaultValue = "")
+    private String binding;
+
+    @Option(names = {"--debug-level"}, defaultValue = "info", description = "Default debug level")
+    private String debugLevel;
+
+    class ShutdownRoute extends RouteBuilder {
+        private File lockFile;
+
+        public ShutdownRoute(File lockFile) {
+            this.lockFile = lockFile;
+        }
+
+        public void configure() {
+            fromF("file-watch://%s?events=DELETE&antInclude=%s", lockFile.getParent(), lockFile.getName())
+                .process(p -> context.shutdown());
+        }
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        switch (debugLevel) {
+            case "trace": Configurator.setRootLevel(Level.TRACE); break;
+            case "debug": Configurator.setRootLevel(Level.DEBUG); break;
+            case "info": Configurator.setRootLevel(Level.INFO); break;
+            case "warn": Configurator.setRootLevel(Level.WARN); break;
+            case "fatal": Configurator.setRootLevel(Level.FATAL); break;
+            default: {
+                System.err.println("Invalid debug level " + debugLevel);
+                return 1;
+            }
+        }
+
+        System.out.println("Starting Camel Jbang!");
+
+        if (binding.isEmpty()) {
+            System.err.println("You must provide a binding file (check --help)");
+
+            return 1;
+        }
+
+        File bindingFile = new File(binding);
+        if (!bindingFile.exists()) {
+            System.err.println("The binding file does not exist");
+
+            return 1;
+        }
+
+        System.setProperty("camel.main.routes-include-pattern", "file:" + binding);
+        System.setProperty("camel.main.name", "CamelJbang");
+
+        KameletMain main = new KameletMain();
+
+        main.configure().addRoutesBuilder(new ShutdownRoute(createLockFile()));
+        main.start();
+        context = main.getCamelContext();
+
+        main.run();
+        return 0;
+    }
+
+    public File createLockFile() throws IOException {
+        File lockFile = File.createTempFile(".run", ".vicuna.lock", new File("."));
+
+        System.out.printf("A new lock file was created on %s. Delete this file to stop running%n",
+            lockFile.getAbsolutePath());
+        lockFile.deleteOnExit();
+
+        return lockFile;
+    }
+}
+
+@Command(name = "search", description = "Run a Kamelet")
```

Comment:
A typo of description ? it should be "Search a Kamelet"

Reply:
You are right! thanks @zhfeng :)